File: src/sensorframe.py
#!/usr/bin/env python

class SensorFrame:

    def __init__(self):
        self.leftWheel = 0
        self.rightWheel= 0
        self.batteryCharge= 0
        self.batteryCapacity = 0
        self.bumperState = 0
        self.lightBumperLeft = 0
        self.lightBumperFrontLeft = 0
        self.lightBumperCenterLeft = 0
        self.lightBumperCenterRight = 0
        self.lightBumperFrontRight = 0
        self.lightBumperRight = 0
        self.oimode = 0

    def isBumperLeft(self):
        return self.bumperState & 2 > 0

    def isBumperRight(self):
        return self.bumperState & 1 > 0

    def isWheeldropLeft(self):
        return self.bumperState & 8 > 0

    def isWheeldropRight(self):
        return self.bumperState & 4 > 0

File: src/sensorframe.cpp
class SensorFrame {

    public:
        int leftWheel;
        int rightWheel;
        int batteryCharge;
        int batteryCapacity;
        int bumperState;
        int lightBumperLeft;
        int lightBumperFrontLeft;
        int lightBumperCenterLeft;
        int lightBumperCenterRight;
        int lightBumperFrontRight;
        int lightBumperRight;
        int oimode;

        SensorFrame() {
            leftWheel = 0;
            rightWheel= 0;
            batteryCharge= 0;
            batteryCapacity = 0;
            bumperState = 0;
            lightBumperLeft = 0;
            lightBumperFrontLeft = 0;
            lightBumperCenterLeft = 0;
            lightBumperCenterRight = 0;
            lightBumperFrontRight = 0;
            lightBumperRight = 0;
            oimode = 0;
        }

        bool isBumperLeft() {
            return (bumperState & 2) > 0;
        }

        bool isBumperRight() {
            return (bumperState & 1) > 0;
        }

        bool isWheeldropLeft() {
            return (bumperState & 8) > 0;
        }

        bool isWheeldropRight() {
            return (bumperState & 4) > 0;
        }
};
